SnailTests/ObservableTests.swift

```diff
@@ -924,4 +925,35 @@ class ObservableTests: XCTestCase {
             subject.on(.next("main - async"))
         }
     }
+
+    func testDisposerRemovesSubscriptionFromObserverOnDeinit() {
```

Comment:
Does this test fail if we undo the changes this PR makes? If so ❤️ , if not lets come up with a use case here!

We should also consider doing this same test for the operators: `.map`, `.merge`, etc.

Reply:
Yes, this test fails without the current changes and passes with them! woo! However, the other `Observable` operators internally initialize a subscriber that gets added to the observable's array of subscribers. Unfortunately I can't think of a good way to add this subscriber to the same disposer as any other subscribers of the observable..